bot/handlers/thread_handler.go

```diff
@@ -78,18 +99,30 @@ func gatewayHandlerThreadMemberUpdate(_ *bot.Client, _ int, _ int, _ gateway.Eve
 func gatewayHandlerThreadMembersUpdate(client *bot.Client, sequenceNumber int, shardID int, event gateway.EventThreadMembersUpdate) {
 	genericEvent := events.NewGenericEvent(client, sequenceNumber, shardID)
 
-	if thread, ok := client.Caches.GuildThread(event.ID); ok {
+	thread, err := client.Caches.GuildThread(event.ID)
+	if err != nil && !errors.Is(err, cache.ErrNotFound) {
+		client.Logger.Error("failed to get thread from cache", slog.Any("err", err), slog.String("thread_id", event.ID.String()))
+	}
+	if err == nil {
 		thread.MemberCount = event.MemberCount
-		client.Caches.AddChannel(thread)
+		if err := client.Caches.AddChannel(thread); err != nil {
+			client.Logger.Error("failed to add channel to cache", slog.Any("err", err), slog.String("thread_id", event.ID.String()))
+		}
 	}
 
 	for _, addedMember := range event.AddedMembers {
 		addedMember.Member.GuildID = event.ID
-		client.Caches.AddThreadMember(addedMember.ThreadMember)
-		client.Caches.AddMember(addedMember.Member)
+		if err := client.Caches.AddThreadMember(addedMember.ThreadMember); err != nil {
+			client.Logger.Error("failed to add thread member to cache", slog.Any("err", err), slog.String("thread_id", event.ID.String()), slog.String("user_id", addedMember.UserID.String()))
+		}
+		if err := client.Caches.AddMember(addedMember.Member); err != nil {
+			client.Logger.Error("failed to add member to cache", slog.Any("err", err), slog.String("guild_id", event.GuildID.String()), slog.String("user_id", addedMember.UserID.String()))
+		}
 
 		if addedMember.Presence != nil {
-			client.Caches.AddPresence(*addedMember.Presence)
+			if err := client.Caches.AddPresence(*addedMember.Presence); err != nil {
+				client.Logger.Error("failed to add presence to cache", slog.Any("err", err), slog.String("guild_id", event.GuildID.String()), slog.String("user_id", addedMember.UserID.String()))
```

Comment:
This isn't very pretty ... There is no way to recover from cache errors here because the whole library is built on top of the assumptions that cache operations never fail ...
Seeing how much noise this already adds to the code I have to wonder if it's actually worth it or if there is a better approach.

Maybe it would be an option to register a central error handler that is called for any cache error so the other parts of the code don't have to handle it? 🤔